src/realm/obj.hpp

```diff
@@ -315,6 +334,11 @@ class Obj : public ConstObj {
     inline void set_spec(T&, ColKey);
 };
 
+struct ConstObj::PathElement {
```

Comment:
I think the path should be independent of a particular group - that is, it should only consist of {table name, ObjectId} to identify the starting point and then a vector of property names - optionally with index.

Reply:
Hmmm. perhaps we have a different use-case in mind here. I expect the replication logic to want to access relevant tables and objects and collect information. This gives good decoupling between Core and the Sync client. And since the code constructing the path would have to visit those same objects and tables, time is saved by simply collecting the accessors along the way.

Reply:
I would not expect that sync need more information. It just want to serialize the path.

Reply:
You can leave it as it is now. But I still hold my position (and will bring it up at any given occasion :smile: )